apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_connector.erl

```diff
@@ -444,19 +444,19 @@ proc_data(
     DataType = list_to_binary(
         string:uppercase(binary_to_list(emqx_placeholder:proc_tmpl(DataType0, Msg)))
     ),
-    case proc_value(DataType, ValueTkn, Msg) of
-        {ok, Value} ->
-            proc_data(T, Msg, Nows, [
-                #{
-                    timestamp => iot_timestamp(TimestampTkn, Msg, Nows),
-                    measurement => emqx_placeholder:proc_tmpl(Measurement, Msg),
-                    data_type => DataType,
-                    value => Value
-                }
-                | Acc
-            ]);
-        Error ->
-            Error
+    try
+        proc_data(T, Msg, Nows, [
+            #{
+                timestamp => iot_timestamp(TimestampTkn, Msg, Nows),
+                measurement => emqx_placeholder:proc_tmpl(Measurement, Msg),
+                data_type => DataType,
+                value => proc_value(DataType, ValueTkn, Msg)
+            }
+            | Acc
+        ])
+    catch
+        _:Reason ->
+            {error, #{hint => invalid_type, reason => Reason}}
```

Comment:
this maybe catches too much.

Turn the old error into throw, e.g. `throw(#{reason => invalid_type}).`
Here 
```
catch throw : Reason ->
          {error, Reason};
      Error : Reason : Stacktrace ->
          ?SLOG(debug, #{exception => Error, reason => Reason, stacktrace => Stacktrace}),
          {error, invalid_data}
```